compose/snippets/src/main/java/com/example/compose/snippets/pictureInPicture/PictureInPictureSnippets.kt

```diff
@@ -0,0 +1,334 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.compose.snippets.pictureInPicture
+
+import android.app.PictureInPictureParams
+import android.app.RemoteAction
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.ContextWrapper
+import android.content.Intent
+import android.content.IntentFilter
+import android.os.Build
+import android.util.Log
+import android.util.Rational
+import androidx.activity.ComponentActivity
+import androidx.annotation.RequiresApi
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.toAndroidRectF
+import androidx.compose.ui.layout.boundsInWindow
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalContext
+import androidx.core.app.PictureInPictureModeChangedInfo
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.toRect
+import androidx.core.util.Consumer
+import androidx.media3.common.Player
+import androidx.media3.exoplayer.ExoPlayer
+
+var shouldEnterPipMode by mutableStateOf(false)
+
+// [START android_compose_pip_broadcast_receiver_constants]
+// Constant for broadcast receiver
+const val ACTION_BROADCAST_CONTROL = "broadcast_control"
+
+// Intent extras for broadcast controls from Picture-in-Picture mode.
+const val EXTRA_CONTROL_TYPE = "control_type"
+const val EXTRA_CONTROL_PLAY = 1
+const val EXTRA_CONTROL_PAUSE = 2
+// [END android_compose_pip_broadcast_receiver_constants]
+
+@Composable
+fun PiPBuilderSetAutoEnterEnabled(
+    modifier: Modifier = Modifier
+) {
+    // [START android_compose_pip_builder_auto_enter]
+    val context = LocalContext.current
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        val pipModifier = modifier.onGloballyPositioned { layoutCoordinates ->
+            val builder = PictureInPictureParams.Builder()
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                builder.setAutoEnterEnabled(true)
+            }
+            context.findActivity().setPictureInPictureParams(builder.build())
+        }
+    } else {
+        Log.i("PiP info", "API does not support PiP")
```

Comment:
nit: Tag should be extracted to a just a const called `private const val PIP_TAG = "PiP info"` seeing as it is used all through this file